hr_contract_amendment/models/hr_contract.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2018 EBII (https://www.saaslys.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    amendment_ids = fields.One2many(
+        comodel_name='hr.contract.amendment', inverse_name="contract_id",
+        string="contract", required=False)
+
+    @api.multi
+    def write(self, vals):
+        if 'job_id' in vals.keys() and len(vals) == 1:
+            type = 'JOB_CHANGE'
+        elif 'wage' in vals.keys() and len(vals) == 1:
+            type = 'SALARY_CHANGE'
+        elif 'working_hours' in vals.keys() and len(vals) == 1:
+            type = 'WORKING_HOURS_CHANGE'
+        else:
+            type = 'CONTRACT_SITUATION'
+
+        EXCLUDED_FIELDS = ['notes']
+        # don't save notes if it's the only one changes
+        if EXCLUDED_FIELDS[0] in vals.keys() and len(vals) == 1:
+            return super(HrContract, self).write(vals)
+        else:
+
+            for res in self:
+                sequence = res.env['ir.sequence'].next_by_code("name")
+                amendmentVals = {
```

Comment:
could you set this code out the function, with the classical _prepare_contract_amendment.
It allows easy overload.